### DISCUSSION TITLE
Create an official Fig plugin and a oh-my-posh theme for sfdx/sf

### DISCUSSION THREAD

**Discussion by AllanOricil** [forcedotcom/cli#2514] [Ideas]
@mshanemc @cromwellryan

obs: fig.io doesn't have a Windows version yet: https://fig.io/user-manual/windows

Now that [Fig](https://fig.io/) is free for everyone, I believe it would be nice if Salesforce could develop and maintain an official Fig plugin. 
I found this one https://github.com/wadewegner/salesforce-cli-zsh-completion but it is outdated. 
Oclif could maybe have a "fig.io" generator plugin, so that all CLIs built with it could automatically generate fig packages (Heroku, for instance). There could exist a new flag in oclif build command to output a fig package as well. During oclif builds, a fig plugin is generated and added to the root of the project.

Look how cool the terminal becomes

https://github.com/forcedotcom/cli/assets/55927613/ad67c499-08b7-46bd-ad6d-d00fe233fafc

Enhanced `cd` plugin! 🥹
<img width="1082" alt="image" src="https://github.com/forcedotcom/cli/assets/55927613/836a64ce-06ef-44a3-883d-664d3aff3a1a">

You could also create a custom theme for [Oh-my-posh](https://ohmyposh.dev/) that displays the default "org alias", [and "group"](https://github.com/forcedotcom/cli/discussions/2444), the cli is configured with. This [theme](https://ohmyposh.dev/docs/themes) would display this information before the git branch

user > cwd -> (sf logo) org alias -> group -> branch

<img width="592" alt="image" src="https://github.com/forcedotcom/cli/assets/55927613/f8cb8677-f9ef-4af8-b921-04072a67a6f7">

This theme can be activated automatically when opening projects with "sfdx-project.json" in the cwd

Look an example where the theme displays the Node version my project is currently configured to. The theme fetches this information from the project and then displays it if available. We could do the same in sfdx!
<img width="689" alt="image" src="https://github.com/forcedotcom/cli/assets/55927613/812fb517-4918-46e9-9dbb-22412e50eb5a">

**Comment by cristiand391** [comment]
I've haven't tried fig yet but they have official oclif plugin:
https://github.com/withfig/oclif-plugin

have you tried it? I'm not sure if it still works (last commit was from ~2y ago).

**Reply by AllanOricil** [reply]
I have not. I see that it depends on oclif@v1, so maybe it no longer works. I will try and tell u later.

**Reply by AllanOricil** [reply]
@cristiand391 

I think this should somehow be part of oclif because when installing a sfdx plugin it will be necessary to recreate fig specs to include new commands.


````js
onPluginInstalled((plugin) => //generate fig spec))
````

I just realized that you can't rely on plugin developers to create fig specs for their plugins. It must be generated by the core package to enforce it on all plugins.

It would be good if clis built with oclif could opt in|out to generate fig specs in case they don't want it.

End user could also generate fig specs in case the cli owner opted to not do so running a command like:

````bash
my_cli fig setup
````

**Comment by AllanOricil** [comment]
🥺

**Reply by cristiand391** [reply]
I don't think this is sf-specific, following from your last comment:

> I just realized that you can't rely on plugin developers to create fig specs for their plugins. It must be generated by the core package to enforce it on all plugins.

What we would need is support for `plugins:postinstall` and `plugins:postuninstall` hooks in [plugin-plugins](https://github.com/oclif/plugin-plugins), then the oclif-fig plugin could re-generate use that to always update the fig specs.
